src/core/dom/index.js

```diff
@@ -507,11 +507,9 @@ export function dispatchCustomEvent(node, name, opt_data, opt_options) {
   const data = opt_data || {};
   // Constructors of events need to come from the correct window. Sigh.
   devAssert(node.ownerDocument);
-  const event = node.ownerDocument.createEvent('Event');
-  event.data = data;
 
   const {bubbles, cancelable} = opt_options || DEFAULT_CUSTOM_EVENT_OPTIONS;
-  event.initEvent(name, bubbles, cancelable);
+  const event = new MessageEvent(name, {data, bubbles, cancelable});
```

Comment:
for my own knowledge, why MessageEvent instead of CustomEvent?

Reply:
based on the usage of `data` it does seem like `MessageEvent` is more appropriate

Reply:
Exactly that :)